internal/pkgecosystem/package.go

```diff
@@ -1,42 +1,50 @@
 package pkgecosystem
 
-type Pkg struct {
-	name    string
-	version string
-	manager *PkgManager
-	local   string
-	command string
+import "fmt"
+
+// Package provides an interface for working with packages from any supported package ecosystem
+type Package interface {
+	// Name returns the name of this package
+	Name() string
+	// Ecosystem returns the ecosystem of this package. It is returned as a string for compatibility reasons,
+	// but is always a valid Ecosystem.
+	Ecosystem() string
+	// Version returns the version of this package
+	Version() string
+	// LocalPath specifies a path to a local archive associated with this package, if one exists.
+	LocalPath() string
+	// Command returns the full command (with arguments) for dynamic analysis
+	Command(phase RunPhase) []string
```

Comment:
Does this have to be part of the interface? It seems like all the implementations just call through your `phaseCommand` function. We can just use that directly instead and avoid a lot of duplication.

Reply:
Good point